File: backend/resource-service/src/main/java/com/gl/resource_service/entity/Resource.java
package com.gl.resource_service.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "resources")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Resource {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long resourceId;

    private Long ownerId;

    private String title;

    private String description;

    private String category;

    private Double rentPerDay;

    private Double securityDeposit;

    private Integer quantity;
    @Enumerated(EnumType.STRING)
    private Condition condition;

    private Boolean available;

    private String city;

    private String state;

    private String imageUrl;

    private LocalDate createdDate;
}

File: frontend/src/types/resource.ts
export type ResourceCondition =
  | "NEW"
  | "LIKE_NEW"
  | "GOOD"
  | "FAIR"
  | "POOR";


export interface ResourceRequest {
  title: string;
  description: string;
  category: string;
  rentPerDay: number;
  securityDeposit: number;
  quantity: number;
  condition: ResourceCondition;
  city: string;
  state: string;
  imageUrl: string;
}


export interface ResourceUpdateRequest {
  title: string;
  description: string;
  category: string;
  rentPerDay: number;
  securityDeposit: number;
  quantity: number;
  condition: ResourceCondition;
  city: string;
  state: string;
  imageUrl: string;
  available: boolean;
}


export interface ResourceResponse {
  resourceId: number;
  ownerId: number;

  title: string;
  description: string;
  category: string;

  rentPerDay: number;
  securityDeposit: number;
  quantity: number;

  condition: ResourceCondition;
  available: boolean;

  city: string;
  state: string;

  imageUrl: string | null;

  createdDate: string;
  
}